pkg/apis/apps/v1alpha1/apicast_types.go

```diff
@@ -16,15 +16,17 @@ type APIcastSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 	// +optional
-	Replicas                  *int64                   `json:"replicas,omitempty"`
-	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef"`
+	Replicas *int64 `json:"replicas,omitempty"`
+	// +optional
+	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef,omitempty"`
+	// +optional
+	GatewayConfigurationSecret *v1.LocalObjectReference `json:"gatewayConfigurationSecret,omitempty"`
```

Comment:
I am not sure what would be a good name for it.
Do you think is there a better name to design "the secret that the user must provide to specify an apicast configuration (json config of apicast) ?

Reply:
Env var is called  `THREESCALE_CONFIG_FILE`

So maybe `ThreeScaleConfigSecret` ???

Reply:
I don't like to have the name ThreeScale where we are already in the context of a 3scale related CR. I understand you have said this name because the envvar is the same but I prefer to making more meaningful / context-related names. The user does not care about envvars but about the effects that they do.

I'll change the name to EmbeddedConfigurationSecret.